src/library.cpp

```diff
@@ -157,6 +157,127 @@ bool Library::removeBookmark(const std::string& zimId, const std::string& url)
   return false;
 }
 
+std::tuple<int, int> Library::migrateBookmarks() {
+  std::set<std::string> sourceBooks;
+  int invalidBookmarks = 0;
+  {
+    std::lock_guard<std::recursive_mutex> lock(m_mutex);
+    for(auto& bookmark:m_bookmarks) {
+      if (m_books.find(bookmark.getBookId()) == m_books.end()) {
+        invalidBookmarks += 1;
+        sourceBooks.insert(bookmark.getBookId());
+      }
+    }
+  }
+  int changed = 0;
+  for(auto& sourceBook:sourceBooks) {
+    changed += migrateBookmarks(sourceBook, ALLOW_DOWNGRADE);
+  }
+  return std::make_tuple(changed, invalidBookmarks);
+}
+
+void Library::removeOlderBooksFromBookCollection(BookIdCollection& books, const Bookmark& bookmark) const {
```

Comment:
This functions, and all functions, should have a documentation.

Reply:
All public methods have a documentation in `include/library.h`.

The particular method has been renamed (and greatly changed) in last version.